moto/mediastoredata/models.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import unicode_literals
+
+import datetime
+from collections import OrderedDict
+
+from boto3 import Session
+
+from moto.core import BaseBackend, BaseModel
+from .exceptions import ClientError
+
+
+class CreatedObject(BaseModel):
+    def __init__(self, *args, **kwargs):
+        self.path = kwargs.get("Path")
+        self.content_sha256 = kwargs.get("ContentSHA256")
+        self.etag = kwargs.get("ETag")
+        self.storage_class = kwargs.get("StorageClass")
+
+    def to_dict(self, exclude=None):
+        data = {
+            "ETag": self.etag,
+            "Name": self.path,
+            "Type": "FILE",
+            "ContentLength": 123,
+            "StorageClass": self.storage_class,
+            "Path": self.path,
+            "ContentSHA256": self.content_sha256,
+        }
+        if exclude:
+            for key in exclude:
+                del data[key]
+        return data
+
+
+class Object(BaseModel):
+    def __init__(self, *args, **kwargs):
+        self.etag = kwargs.get("ETag")
+        self.content_type = kwargs.get("ContentType")
+        self.content_length = kwargs.get("ContentLength")
+        self.cache_control = kwargs.get("CacheControl")
+        self.last_modified = datetime.datetime.now()
+
+    def to_dict(self, exclude=None):
+        data = {
+            "ETag": self.etag,
+            "ContentType": self.content_type,
+            "ContentLength": self.content_length,
+            "CacheControl": self.cache_control,
+            "LastModified": self.last_modified,
+        }
+        if exclude:
+            for key in exclude:
+                del data[key]
+        return data
+
+
+class MediaStoreDataBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(MediaStoreDataBackend, self).__init__()
+        self.region_name = region_name
+        self._objects = OrderedDict()
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    def put_object(
+        self,
+        Body,
+        Path,
+        ContentType=None,
+        CacheControl=None,
+        StorageClass="TEMPORAL",
+        UploadAvailability="STANDARD",
+    ):
+        new_object = CreatedObject(
+            Path=Path, ContentSHA256=Body, ETag="etag", StorageClass="TEMPORAL"
```

Comment:
The `body` and sha256 should be stored separately. Probably easiest to only pass the body, and let the `CreatedObject` figure out the hash-value (`hashlib.sha256(body).hexdigest()`)